parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -1111,29 +1115,31 @@ pushGitRootBranch srcConn branch repo = runExceptT @C.GitError do
     lift . flip State.execStateT emptySyncProgressState $
       syncInternal syncProgress srcConn destConn (Branch.transform lift branch)
     flip runReaderT destConn do
-      let newRootHash = Branch.headHash branch
-      -- the call to runDB "handles" the possible DB error by bombing
-      (fmap . fmap) Cv.branchHash2to1 (runDB destConn Ops.loadMaybeRootCausalHash) >>= \case
-        Nothing -> do
-          setRepoRoot newRootHash
-          Q.release "push"
-        Just oldRootHash -> do
-          before oldRootHash newRootHash >>= \case
-            Nothing ->
-              error $
-                "I couldn't find the hash " ++ show newRootHash
-                  ++ " that I just synced to the cached copy of "
-                  ++ repoString
-                  ++ " in "
-                  ++ show remotePath
-                  ++ "."
-            Just False -> do
-              Q.rollbackRelease "push"
-              throwError . C.GitProtocolError $ GitError.PushDestinationHasNewStuff repo
-
-            Just True -> do
+      if setRoot
```

Comment:
The change is hard to see because of indentation.

Before:

```
if behind then complain else go ahead
```

After:

```
if setRoot then
  if behind then complain else go ahead -- as before
else go ahead -- not setting remote root; nothing to check
```